databricks/koalas/groupby.py

```diff
@@ -2162,7 +2169,9 @@ def pandas_transform(pdf):
                 retain_index=False,
             )
             # Otherwise, it loses index.
```

Comment:
I may not understand `# Otherwise, ` in this context. I was thinking `transform` would always lose the index.